scalding-core/src/main/scala/com/twitter/scalding/ExecutionUtil.scala

```diff
@@ -0,0 +1,92 @@
+package com.twitter.scalding
+
+import java.util.concurrent.Semaphore
+
+import com.twitter.algebird.Monoid
+import scala.util.{ Failure, Success }
+
+object ExecutionUtil {
+  /**
+   * Run a sequence of executions but only permitting parallelism amount to run at the
+   * same time.
+   *
+   * @param executions List of executions to run
+   * @param parallelism Number to run in parallel
+   * @return Execution Seq
+   */
+  def withParallelism[T](executions: Seq[Execution[T]], parallelism: Int): Execution[Seq[T]] = {
+    require(parallelism > 0, s"Parallelism must be > 0: $parallelism")
+
+    val lock = new Semaphore(parallelism)
+
+    def waitRun(e: Execution[T]): Execution[T] = {
+      Execution.from(lock.acquire())
```

Comment:
Let me see what I can come up with without the complexity and added dependencies of pulling in twitter futures etc.  I was trying to avoid that above but you are correct about the blocking.  I agree about the unit tests, hard to think of a simple fast test that can exercise the parallelism portion of the code.